ducc/src/ducc.rs

```diff
@@ -274,6 +274,53 @@ impl Ducc {
         }
     }
 
+    /// Returns the callstack entry at given level.
+    /// 
+    /// # Levels
+    /// 
+    /// - `-1` is the most recent function called. If the inspect function is invoked from inside
+    ///     a Ducc callback function, this will be the function callback itself.
+    /// - `-2` is the caller of `-1`
+    /// - and so on..
+    /// 
+    /// # Example
+    /// 
+    /// ```
+    /// # use ducc::Ducc;
+    /// # let ducc = Ducc::new();
+    /// if let Some(entry) = ducc.inspect_callstack_entry(-2) {
+    ///    let file_name = entry.function.into_object().get::<_, String>("fileName").unwrap();
+    ///    println!("Function was called from {} line {}", file_name, entry.line_number);
+    /// };
+    /// ```
+    pub fn inspect_callstack_entry<'ducc>(&'ducc self, level: i32) -> Option<CallStackEntry<'ducc>> {
```

Comment:
This could be a `Result`, but would it be worth it for a new `ErrorKind` just for calling the function with invalid level?

Reply:
I think this API is good as-is! I would just add a note to the doc comment that `None` is returned if the given level goes beyond the current callstack depth.